src/files/BrsFile.ts

```diff
@@ -121,14 +134,93 @@ export class BrsFile {
         }
     }
 
-
-    public parser: Parser;
+    public parser = new Parser();
 
     public fileContents: string;
 
+    private logDebug(...args) {
+        this.program.logger.debug('BrsFile', chalk.green(this.pkgPath), ...args);
+    }
+
+    /**
+     * If this is a typedef file
+     */
+    public isTypedef: boolean;
+
+    /**
+     * The key to find the typedef file in the program's files map.
+     * A falsey value means this file is ineligable for a typedef
+     */
+    public typedefKey?: string;
+
+    /**
+     * If the file was given type definitions during parse
+     */
+    public get hasTypedef() {
+        return !!this.typedefFile;
+    }
+
+    /**
+     * A reference to the typedef file (if one exists)
+     */
+    public get typedefFile() {
+        if (this.typedefKey) {
+            return this.cache.getOrAdd('typedefFile', () => {
+                const file = this.program.getFileByPathAbsolute<BrsFile>(this.typedefKey);
+                return file;
+            });
+        }
+    }
```

Comment:
This type of code should have an explicit `return undefined` fallback.

Reply:
done.